python/mpt/src/mpt/trie_diff.cairo

```diff
@@ -1704,14 +1922,15 @@ func _compute_left_extension_node_diff_on_right_branch_node{
     // case we compare to the subnode
     // 2. It doesn't in which case we compare to null.
 
-    let l_extension = left.value.extension;
+    let l_extension = left.value.extension_node;
     let first_nib = l_extension.value.key_segment.value.data[0];
     if (first_nib == index) {
         // Fully consumed by this nibble: compare to the subnode
         if (l_extension.value.key_segment.value.len == 1) {
             let node_to_compare_ = OptionalUnionInternalNodeExtendedImpl.from_extended(
                 l_extension.value.subnode
             );
+            tempvar current_parent_left = left;
```

Comment:
now that I think of it it would be "cleaner" to define 

```
local node_to_compare: OptionalUnionInternalNodeExtended;
local current_parent_left: OptionalInternalNode;
```

and doing `assert node_to_compare = xxx` in each branch rather than having tempvar that we need to re-bind at the end

Reply:
Ok!